src/App.tsx

```diff
@@ -1,7 +1,83 @@
-import { greet } from "./utils/greet";
-
+import episodes from "./episodes.json";
+import EpisodesDisplay from "./Components/EpisodesDisplay";
+import { useEffect, useState } from "react";
+import { isMatching } from "./utils/matchingEpisodes";
+import { IEpisode } from "./utils/interface";
 function App(): JSX.Element {
-  return <h1>{greet("World")}</h1>;
+  const [searchTerm, setSearchTerm] = useState("");
+  console.log("inside the App function");
+  const [episodesData, setEpisodesData] = useState<IEpisode[]>([]);
+  useEffect(() => {
+    const getAndStoreDataFromApi = async () => {
+      const response = await fetch(
+        `https://api.tvmaze.com/shows/${episodes[0].id}/episodes`
+      );
+      const fetchedEpisodes: IEpisode[] = await response.json();
+      setEpisodesData(fetchedEpisodes);
+    };
+    getAndStoreDataFromApi();
+  }, []);
+  async function handleOnChange(e: React.ChangeEvent<HTMLSelectElement>) {
+    //console.log(`https://api.tvmaze.com/shows/${e.target.value}/episodes`);
+    const response = await fetch(
+      `https://api.tvmaze.com/shows/${e.target.value}/episodes`
+    );
+    const fetchedEpisodes: IEpisode[] = await response.json();
+    setEpisodesData(fetchedEpisodes);
+    setSearchTerm("");
+  }
+  const showsSorted = episodes.sort(function (x, y) {
+    if (x.name < y.name) return -1;
+    if (x.name > y.name) return 1;
+    return 0;
+  });
+  const filteredEpisodes = isMatching(episodesData, searchTerm);
+  async function handleOnChangeEpisode(
+    e: React.ChangeEvent<HTMLSelectElement>
+  ) {
+    setSearchTerm(e.target.value);
+  }
+  return (
+    <>
+      <div style={{ display: "flex", margin: "10px" }}>
+        <select onChange={handleOnChange}>
+          {showsSorted.map((show) => (
+            <option key={show.id} value={show.id}>
+              {show.name}
+            </option>
+          ))}
+        </select>
+        <select onChange={handleOnChangeEpisode}>
+          {filteredEpisodes.map((episode) => (
+            <option key={episode.id} value={episode.name}>
+              {episode.name}
+            </option>
+          ))}
+        </select>
+        <input
+          value={searchTerm}
+          onChange={(e) => {
+            setSearchTerm(e.target.value);
+          }}
+        />
+        <p>
+          {" "}
+          Displaying {filteredEpisodes.length}/{episodesData.length} episodes{" "}
+        </p>
+      </div>
+      <div style={{ display: "flex", flexWrap: "wrap" }}>
+        {filteredEpisodes.map((element, index) => (
+          <EpisodesDisplay key={index} {...element} />
+        ))}
+      </div>
+      <div>
+        <p>https://api.tvmaze.com/shows/{parseFloat(searchTerm)}/episodes</p>
```

Comment:
parseFloat argument is a string and returns a number, but the arguments should have a number in the string otherwise it would return NaN (as shown in the deployed app)